### DISCUSSION TITLE
Bootstrap for BitBucket as Git provider

### DISCUSSION THREAD

**Discussion by haraldatbmw** [fluxcd/flux2#377] [Proposals]
Are there any plans to add the same great bootstrap functionality for BitBucket as a Git provider too?
I know the manual way but would love to see the full automated comfort feature :-)

**Comment by ghost** [comment]
We are also heavily using bitbucket in our company so that would be really cool

**Comment by stefanprodan** [comment]
We (the flux2 maintainers) have no plans on implementing BitBucket any time soon. I looked a little into BitBucket APIs, the SaaS one uses a different API to the on-prem one, implementing this across SaaS/on-prem is not a trivial task. If someone that has experience with BitBucket APIs and Go wants to help, please let us know.

**Reply by ghost** [reply]
Thanks for the reply. We can work around with the flux install command like described in the documentation on generic git server.

**Reply by ramzan118** [reply]
I am facing issue of bootstraping for flux and bitbucket. can you help in this regard? flux bootstrap bitbucket   --owner=localgitopssetup   --repository=gitops   --branch=techworks2023/backgammon   --path=clusters/kind-multi-node-cluster   --personal
✗ unknown flag: --owner

**Comment by haraldatbmw** [comment]
As a workaround we use terraform instead of flux cli for bootstrapping with bitbucket. 
Terraform scripts are located inside our BitBucket ops-repository and do the following:

1. Create or overwrite gotk-components.yaml, gotk-sync.yaml, kustomization.yaml into flux-system folder
2. Use kustomize to generate patched manifests
3. Apply patched manifests with kubectl to k8s cluster
4. Commit & Push file in folder flux-system to BitBucket

**Reply by ghost** [reply]
Thank for the info! We will do a similar approach too now.

**Reply by saiyen2002** [reply]
@haraldatbmw  do you think you could ever share those scripts? it would be very helpful

**Comment by stefanprodan** [comment]
In Flux [0.24.0](https://github.com/fluxcd/flux2/releases/tag/v0.24.0) we've added support for **BitBucket Server and Data Center** to bootstrap, docs here: https://fluxcd.io/docs/installation/#bitbucket-server-and-data-center
